share/CMakeJS.cmake

```diff
@@ -0,0 +1,642 @@
+endif()
+
+#[=============================================================================[
+Get the in-use NodeJS binary for executing NodeJS commands in CMake scripts.
+
+Provides
+
+::
+
+  NODE_EXECUTABLE, the NodeJS runtime binary being used
+  NODE_VERSION, the version of the NodeJS runtime binary being used
+
+]=============================================================================]#
+function(cmakejs_acquire_node_executable)
+    find_program(NODE_EXECUTABLE
+      NAMES "node" "node.exe"
+      PATHS "$ENV{PATH}" "$ENV{ProgramFiles}/nodejs"
+      DOC "NodeJs executable binary"
+      REQUIRED
+    )
+    if (NOT NODE_EXECUTABLE)
+        message(FATAL_ERROR "NodeJS installation not found! Please check your paths and try again.")
+        return()
+    endif()
+
+    execute_process(
+      COMMAND "${NODE_EXECUTABLE}" "--version"
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      OUTPUT_VARIABLE NODE_VERSION
+    )
+    string(REGEX REPLACE "[\r\n\"]" "" NODE_VERSION "${NODE_VERSION}")
+    set(NODE_VERSION "${NODE_VERSION}" CACHE STRING "" FORCE)
+
+    if(VERBOSE)
+        message(STATUS "NODE_EXECUTABLE: ${NODE_EXECUTABLE}")
+        message(STATUS "NODE_VERSION: ${NODE_VERSION}")
+    endif()
+endfunction()
+
+if(NOT DEFINED NODE_EXECUTABLE)
+    cmakejs_acquire_node_executable()
+    message(STATUS "NODE_EXECUTABLE: ${NODE_EXECUTABLE}")
+    message(STATUS "NODE_VERSION: ${NODE_VERSION}")
+endif()
+
+# Resolve NodeJS development headers
+# TODO: This code block is quite problematic, since:
+# 1 - it might trigger a build run, depending on how the builder has set up their package.json scripts...
+# 2 - it also currently assumes a preference for yarn over npm (and the others)...
+# 3 - finally, because of how cmake-js works, it might create Ninja-build artefacts,
+# even when the CMake user specifies a different generator to CMake manually...
+# We could use 'add_custom_target()' with a user-side ARG for which package manager to use...
+if(NOT IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/node_modules")
+    execute_process(
+      COMMAND yarn install
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      OUTPUT_VARIABLE NODE_MODULES_DIR
+    )
+    if(NOT IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/node_modules")
+        message(FATAL_ERROR "Something went wrong - NodeJS modules installation failed!")
+        return()
+    endif()
+endif()
+
+#[=============================================================================[
+Get NodeJS C Addon development files.
+
+Provides
+::
+
+  NODE_API_HEADERS_DIR, where to find node_api.h, etc.
+  NODE_API_INC_FILES, the headers required to use Node Addon API.
+
+]=============================================================================]#
+function(cmakejs_acquire_napi_c_files)
+    execute_process(
+      COMMAND "${NODE_EXECUTABLE}" -p "require('node-api-headers').include_dir"
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
```

Comment:
I just thought, this should be searching from maybe `"${CMAKE_CURRENT_SOURCE_DIR}/node_modules/cmake-js"`, as this is installed as a dependency of cmake-js rather than the users project. I have a feeling that some package managers will not be able to find the dependency if searching from the user project directory